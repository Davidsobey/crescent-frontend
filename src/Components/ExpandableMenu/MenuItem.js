import React from 'react';
import PropTypes from 'prop-types';
import List, { ListItem, ListItemIcon, ListItemText } from 'material-ui/List';
import { withStyles } from 'material-ui/styles';
import { Link } from 'react-router-dom';

const styles = theme => ({
  nested: {
    paddingLeft: theme.spacing.unit * 4,
  },
  accentColor: {
    color: 'white',
    fontSize: '14px',
  },
});

function MenuItem(props) {
  const {
    target, option, classes, icon, extention,
  } = props;
  return (
    <List disablePadding>
      <Link to={`/${target.toLowerCase()}/${extention}`}>
        <ListItem button className={classes.nested}>
          <ListItemIcon className={classes.accentColor}>{icon}</ListItemIcon>
          <ListItemText
            className={classes.accentColor}
            inset
            disableTypography
            primary={option}
          />
        </ListItem>
      </Link>
    </List>
  );
}

MenuItem.propTypes = {
  target: PropTypes.string.isRequired,
  option: PropTypes.string.isRequired,
  extention: PropTypes.string.isRequired,
  icon: PropTypes.object.isRequired,
  classes: PropTypes.object.isRequired,
};

export default withStyles(styles)(MenuItem);
