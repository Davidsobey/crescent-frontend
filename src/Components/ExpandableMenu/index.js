import React from 'react';
import PropTypes from 'prop-types';

import { ListItem, ListItemIcon, ListItemText } from 'material-ui/List';
import { withStyles } from 'material-ui/styles';
import ExpandMore from 'material-ui-icons/ExpandMore';
import ExpandLess from 'material-ui-icons/ExpandLess';
import Collapse from 'material-ui/transitions/Collapse';

import MenuItem from './MenuItem';

const styles = {
  accentColor: {
    color: 'white',
    fontSize: '14px',
  },
};

class ExpandableMenu extends React.Component {
  state = { menuItem: false };
  handleClick = () => {
    const newItem = this.props.details.listName;

    if (newItem && newItem !== this.state.menuItem) {
      this.setState({ menuItem: newItem });
    }

    if (!newItem || newItem === this.state.menuItem) {
      this.setState({ menuItem: false });
    }
  };

  render() {
    const { details, classes } = this.props;

    return (
      <div>
        <ListItem button onClick={this.handleClick}>
          <ListItemIcon className={classes.accentColor}>
            {details.listIcon}
          </ListItemIcon>
          <ListItemText
            inset
            className={classes.accentColor}
            disableTypography
            primary={details.listName}
          />
          {this.state.menuItem === details.listName ? (
            <ExpandLess className={classes.accentColor} />
          ) : (
            <ExpandMore className={classes.accentColor} />
          )}
        </ListItem>
        <Collapse
          component="li"
          in={this.state.menuItem === details.listName}
          timeout="auto"
          unmountOnExit
        >
          {details.subItems.map(item => (
            <MenuItem
              key={item.key}
              target={details.listName}
              option={item.subItemName}
              icon={item.subItemIcon}
              extention={item.subItemExtension}
            />
          ))}
        </Collapse>
      </div>
    );
  }
}

ExpandableMenu.propTypes = {
  classes: PropTypes.object.isRequired,
  details: PropTypes.object.isRequired,
};

export default withStyles(styles)(ExpandableMenu);
