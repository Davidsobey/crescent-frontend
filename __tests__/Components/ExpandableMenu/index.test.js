import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import configureStore from 'redux-mock-store'; // Smart components

// Component to be tested
import ExpandableMenu from '../../../src/Components/ExpandableMenu/index.js';

import AddIcon from 'material-ui-icons/Add';
import ListIcon from 'material-ui-icons/ViewList';
import CourseIcon from 'material-ui-icons/ChromeReaderMode';

const mockStore = configureStore();
const initialState = {};
const store = mockStore(initialState);

const CourseDetails = {
  listName: 'Course',
  listIcon: <CourseIcon />,
  subItems: [
    {
      key: 0,
      subItemName: 'Create Course',
      subItemIcon: <AddIcon />,
      subItemExtension: 'create',
    },
    {
      key: 1,
      subItemName: 'Course List',
      subItemIcon: <ListIcon />,
      subItemExtension: 'list',
    },
  ],
};

describe('<ExpandableMenu />', () => {
  describe('render()', () => {
    test('renders the component', () => {
      const wrapper = shallow(<ExpandableMenu store={store} details={CourseDetails} />);
      const component = wrapper.dive();

      expect(toJson(component)).toMatchSnapshot();
    });
  });
});