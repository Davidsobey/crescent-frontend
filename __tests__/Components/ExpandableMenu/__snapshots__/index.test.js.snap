// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ExpandableMenu /> render() renders the component 1`] = `
<div>
  <WithStyles(ListItem)
    button={true}
    onClick={[Function]}
  >
    <WithStyles(ListItemIcon)
      className="ExpandableMenu-accentColor-1"
    >
      <pure(ChromeReaderMode) />
    </WithStyles(ListItemIcon)>
    <WithStyles(ListItemText)
      className="ExpandableMenu-accentColor-1"
      disableTypography={true}
      inset={true}
      primary="Course"
    />
    <pure(ExpandMore)
      className="ExpandableMenu-accentColor-1"
    />
  </WithStyles(ListItem)>
  <WithStyles(Collapse)
    component="li"
    in={false}
    timeout="auto"
    unmountOnExit={true}
  >
    <WithStyles(MenuItem)
      extention="create"
      icon={<pure(Add) />}
      key="0"
      option="Create Course"
      target="Course"
    />
    <WithStyles(MenuItem)
      extention="list"
      icon={<pure(ViewList) />}
      key="1"
      option="Course List"
      target="Course"
    />
  </WithStyles(Collapse)>
</div>
`;
